### PR TITLE
feat(tooltip-popover-combo): add new component

### PR DESCRIPTION
**Pull request by jor-row**
# Description

This creates a component that allows you to have a tooltip and a popover on the same trigger component. The tooltip and popover cannot be open at the same time.

# Links

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-525737

### CODE REVIEW

File: src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -23,17 +23,29 @@ const Tooltip = forwardRef(
       [isDescription]
     );
 
+    // Modified tooltipSetInstance to call both setInstance and updateInstance
+    const setTooltipInstance = useCallback(
+      (popoverInstance: PopoverInstance | undefined) => {
+        if (otherProps?.setInstance) {
+          otherProps.setInstance(popoverInstance);
+        }
```
Comment (maxinteger) [inline src/components/Tooltip/Tooltip.tsx:31]: I do not think we need the extra `useCallback` here.

If you rename `updateInstance` to `setInstance` then you do it there as well

also you can make it in a single line:
```suggestion
        otherProps?.setInstance?.(popoverInstance);
```

File: src/components/TooltipPopoverCombo/TooltipPopoverCombo.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  const [tooltipInstance, setTooltipInstance] = useState<PopoverInstance>();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [isTooltipOpen, setIsTooltipOpen] = useState(false);
+
+  // Modified tooltipSetInstance to call both setInstance and updateInstance
+  const setMergedTooltipInstances = useCallback(
+    (popoverInstance: PopoverInstance | undefined) => {
+      if (otherTooltipProps?.setInstance) {
+        otherTooltipProps?.setInstance(popoverInstance);
+      }
```
Comment (maxinteger) [inline src/components/TooltipPopoverCombo/TooltipPopoverCombo.tsx:29]: nitpicking:

```suggestion
      otherTooltipProps?.setInstance?.(popoverInstance);
```

and the same for the rest

**Review by maxinteger — APPROVED** [review]
Looks good

File: src/components/TooltipPopoverCombo/TooltipPopoverCombo.tsx (hunk 2 above)
Comment (jor-row) [reply src/components/TooltipPopoverCombo/TooltipPopoverCombo.tsx:29]: made suggested change

**Pull request merged**